src/vision/status_quo/barbara_polls_a_mutex.md

```diff
@@ -0,0 +1,151 @@
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## Barbara polls a Mutex
+
+### Brief summary
+
+[Barbara] is implementing an interpreter for a scripting language. This
+language has implicit asynchronicity, so all values in the language can
+potentially be futures underneath.
+
+Barbara wants to store a namespace which maps variable names to their values. She
+chooses to use a `HashMap` and finds the `async_lock` crate provides an async
+mutex, which she can use for concurrency. She determines she'll need a lock
+around the namespace itself to protect concurrent modification.
+
+For the entries in her map, Barbara decides to implement a two-variant enum. One
+variant indicates that there is no implicit asynchronicity to resolve and the
+value is stored directly here. The other variant indicates that this value is
+being computed asynchronously and polling will be required to resolve it.
+Because an asynchronous task might want to change one of these entries from the
+asynchronous variant to the ready variant, she'll need to wrap the entries in
+an `Arc` and a `Mutex` to allow an asynchronous task to update them.
+
+Barbara wants to be able to derive a future from the entries in her namespace
+that will allow her to wait until the entry becomes ready and read the value.
+She decides to implement the `Future` trait directly. She's done this before
```

Comment:
It's unclear to me why implementing `Future` directly is desirable. Can you talk a bit more about why Barbara did not use async/await notation?